### PR TITLE
Update php unit

### PR DESCRIPTION
**Pull request by Yozhef**

### CODE REVIEW

**Review by shadowhand — CHANGES_REQUESTED** [review]
There's a mix of things happening here. Is the goal to add types, or to upgrade dependencies?

**Comment by Yozhef** [comment]
this branch checkout from the branch when Update return type. after merging return type, only the update deps

**Comment by shadowhand** [comment]
@Yozhef needs rebase.

**Comment by Yozhef** [comment]
@shadowhand can you add the setting in repo -> update Branch if my branch not lasted master. we use it successfully on projects then it is not necessary to do many manipulations locally

File: test/src/Provider/HostedDomainCheckTest.php [hunk 1]
```diff
@@ -65,9 +71,11 @@ public function validHostedDomainProvider()
      * Test combinations of hosted domain and user data that are invalid
      *
      * @dataProvider invalidHostedDomainProvider
-     * @param $json
+     *
+     * @param array  $providerConfig
+     * @param string $json
```
Comment (shadowhand) [inline test/src/Provider/HostedDomainCheckTest.php:76]: No need to duplicate typed parameters.

File: test/src/Provider/HostedDomainCheckTest.php [hunk 2]
```diff
@@ -3,26 +3,32 @@
...
+     * @param string|null $expectedHostedDomain
+     *
+     * @throws Exception
      */
-    public function testValidHostedDomains(array $providerConfig, $json, $expectedHostedDomain)
-    {
+    public function testValidHostedDomains(
+        array $providerConfig,
+        string $json,
+        ?string $expectedHostedDomain
```
Comment (shadowhand) [inline test/src/Provider/HostedDomainCheckTest.php:30]: Nullable parameters should have a `= null` default.

**Review by shadowhand — CHANGES_REQUESTED** [review]

**Review by shadowhand — APPROVED** [review]

**Comment by shadowhand** [comment]
@Yozhef thanks for all these PRs! Not sure when the release will be.

**Comment by Yozhef** [comment]
@shadowhand And we can make a release after adding support for php 8.0 - I just need it in the project in 5 days?

**Comment by shadowhand** [comment]
@Yozhef yes, I think I will tag a new major version after adding PHP 8 support.

**Pull request merged**